addons/slack/install.js

```diff
@@ -13,64 +13,97 @@ const getTemplateFiles = (fileNames) =>
-  .sort()
-  .map((k) => `${k}=${storageSpecification[k]}`)
-  .join('\n')}
 `,
-      nodejs: {
-        files: getTemplateFiles(['index.js', 'package.json', 'slack.js', 'health.js']),
-      },
-      metadata: {
-        fusebit: {
-          editor: {
-            navigationPanel: {
-              hideFiles: ['index.js'],
+          nodejs: {
+            files: getTemplateFiles(['index.js', 'package.json', 'slack.js', 'health.js']),
+          },
+
+          security: {
+            // External calls from customer infrastructure
+            // Internal calls from connector instance with Events API messages
+            authentication: 'optional',
+            authorization: [
+              {
+                action: 'function:execute',
+                resource:
+                  '/account/{{accountId}}/subscription/{{subscriptionId}}/boundary/{{boundaryId}}/function/{{functionId}}/',
+              },
+            ],
+            functionPermissions: {
+              allow: [
+                ...(eventsResourcePath
+                  ? [
+                      {
+                        // :execute connector instance
+                        action: 'function:execute',
+                        resource: eventsResourcePath,
+                      },
+                    ]
+                  : []),
+                {
+                  // :post to its own storage
+                  action: 'storage:*',
+                  resource:
+                    '/account/{{accountId}}/subscription/{{subscriptionId}}/storage/boundary/{{boundaryId}}/function/{{functionId}}/',
+                },
+              ],
+            },
+          },
+          metadata: {
+            fusebit: {
+              editor: {
+                navigationPanel: {
+                  hideFiles: ['index.js'],
+                },
+              },
             },
+            ...ctx.body.metadata,
           },
         },
-        ...ctx.body.metadata,
-      },
-    });
-  } catch (e) {
-    // Could not create function, clean up storage
-    await Sdk.deleteStorage(ctx, storageSpecification);
-    throw e;
-  }
-
-  let registrationUrl = ctx.body.configuration.fusebit_events_registration;
-  let token = ctx.body.configuration.fusebit_events_token;
-  if (registrationUrl && token) {
-    try {
-      Sdk.debug('Events API configuration was present');
-      await Superagent.post(registrationUrl)
-        .set('Authorization', `Bearer ${token}`)
-        .send({
-          app: ctx.body.configuration.slack_app_id,
-          team_id: ctx.body.configuration.slack_team_id,
-          handler: (await Sdk.getFunctionUrl(ctx)) + '/events/ingest',
-        });
-
-      Sdk.debug('Events API registration succeeded');
+        ctx.fusebit.functionAccessToken
+      );
     } catch (e) {
-      // Couldn't do Events API registration, clean up function and storage
-      await Sdk.deleteStorage(ctx, storageSpecification);
-      await Sdk.deleteFunction(ctx);
       throw e;
```

Comment:
If it’s just a catch/throw, then it probably can be removed.

Reply:
Good catch, fixed